lib/probes/morgan.js

```diff
@@ -1,135 +1,42 @@
 'use strict'
 
 const ao = require('..')
+const shimmer = require('shimmer')
 
 const logMissing = ao.makeLogMissing('morgan')
 
-let token
-
-//
-// patch morgan so the arguments can be mucked with. this is a proxy
-// because 1) morgan is typically only called once or maybe twice and
-// 2) it will let us intercept token and format calls at some point if
-// we choose to do so.
-//
-module.exports = function (morgan) {
+module.exports = function (morgan, info) {
```

Comment:
Random curiosity, what is defining / providing `info` and how is it used?

Reply:
It’s the signature of the require patch. Info is an object containing name and version (https://github.com/appoptics/appoptics-apm-node/blob/0b0591e61c598bc33e4bb27050012505b546e652/lib/require-patch.js#L98)

it’s used when probe needs variation in relation to original package version (e.g. https://github.com/appoptics/appoptics-apm-node/blob/0b0591e61c598bc33e4bb27050012505b546e652/lib/probes/pino.js#L25)

Reply:
I see, thanks :)